openbas-front/src/admin/components/simulations/simulation/ExerciseDateForm.tsx

```diff
@@ -0,0 +1,188 @@
+import { z } from 'zod';
+import type { ExerciseUpdateStartDateInput } from '../../../../utils/api-types';
+import { useFormatter } from '../../../../components/i18n';
+import { zodImplement } from '../../../../utils/Zod';
+import { minutesInFuture } from '../../../../utils/Time';
+
+interface Props {
+  onSubmit: SubmitHandler<ExerciseUpdateStartDateInput>;
+  initialValues?: ExerciseUpdateStartDateInput;
+  editing?: boolean;
+  handleClose: () => void;
+}
+
+interface ExerciseStartDateAndTime {
+  date: string;
+  time: string;
+}
+
+// eslint-disable-next-line no-underscore-dangle
+const _MS_DELAY_TOO_CLOSE = 1000 * 60 * 2;
+
+const defaultFormValues = () => ({
+  date: new Date(new Date().setUTCHours(0, 0, 0, 0)).toISOString(),
+  time: minutesInFuture(5).toISOString(),
+});
+
+const ExerciseDateForm: React.FC<Props> = ({
+  onSubmit,
+  handleClose,
+  initialValues,
+}) => {
+  const { t } = useFormatter();
+
+  const [checked, setChecked] = useState(true);
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setChecked(event.target.checked);
+  };
+
+  const submit = (data: ExerciseStartDateAndTime) => {
+    if (checked) {
+      onSubmit({ exercise_start_date: '' });
+    } else {
+      const date = new Date(data.date);
+      const time = new Date(data.time);
+      date.setHours(time.getHours());
+      date.setMinutes(time.getMinutes());
+      date.setSeconds(time.getSeconds());
+      onSubmit({ exercise_start_date: date.toISOString() });
+    }
+  };
+
+  const {
+    control,
+    handleSubmit,
+    clearErrors,
+    getValues,
+    reset,
+  } = useForm<ExerciseStartDateAndTime>({
+    defaultValues: defaultFormValues(),
+    resolver: zodResolver(
+      zodImplement<ExerciseStartDateAndTime>().with({
+        date: z.string().min(1, t('Required')),
+        time: z.string().min(1, t('Required')),
+      }).refine(
+        (data) => {
+          if (data.date) {
+            return new Date(data.date).getTime() >= new Date(new Date().setUTCHours(0, 0, 0, 0)).getTime();
+          }
+          return true;
+        },
+        {
+          message: t('Date should be at least today'),
+          path: ['date'],
+        },
+      )
+        .refine(
+          (data) => {
+            if (data.time) {
+              return new Date(new Date().setUTCHours(0, 0, 0, 0)).getTime() !== new Date(data.date).getTime()
+                || (new Date().getTime() + _MS_DELAY_TOO_CLOSE) < new Date(data.time).getTime();
+            }
+            return true;
+          },
+          {
+            message: t('The time and start date do not match, as the time provided is either too close to the current moment or in the past'),
+            path: ['time'],
+          },
+        ),
+    ),
+  });
+
+  useEffect(() => {
+    if (initialValues?.exercise_start_date != null) {
+      reset({
+        date: initialValues.exercise_start_date,
+        time: initialValues.exercise_start_date,
+      });
+    }
+  });
```

Comment:
When my simulation has not start date, the dialog has the manual launch option checked -> OK
When my simulation has a start date, the dialog has the manual launch option checked -> NOK

Should have the good property set
![image](https://github.com/OpenBAS-Platform/openbas/assets/53513584/a980298c-b590-446f-a9f3-3b07c969f8fa)

Reply:
Bug found
When I change the Time, is not updated